From dd3e2f4d9be661a5fae96c81618fb25c438916da Mon Sep 17 00:00:00 2001
From: Dan Winship <danw@gnome.org>
Date: Tue, 10 Dec 2013 10:34:08 +0100
Subject: [PATCH] SoupSocket: set TCP_NODELAY on sockets

The old SoupSocket did this but it accidentally got removed in the
GSocket port because I thought it didn't matter. It does.

Noticed by Andy Wingo.
---
 libsoup/soup-socket.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libsoup/soup-socket.c b/libsoup/soup-socket.c
index c101fa5..e0748fd 100644
--- a/libsoup/soup-socket.c
+++ b/libsoup/soup-socket.c
@@ -182,6 +182,7 @@ finish_socket_setup (SoupSocketPrivate *priv)
 		priv->ostream = g_object_ref (g_io_stream_get_output_stream (priv->iostream));
 
 	g_socket_set_timeout (priv->gsock, priv->timeout);
+	g_socket_set_option (priv->gsock, IPPROTO_TCP, TCP_NODELAY, TRUE, NULL);
 }
 
 static void
-- 
1.8.5.3

